gms/gossiper.cc

```diff
@@ -814,7 +814,7 @@ gossiper::endpoint_lock_entry::endpoint_lock_entry() noexcept
     , pid(permit_id::create_null_id())
 {}
 
-future<gossiper::endpoint_permit> gossiper::lock_endpoint(inet_address ep, permit_id pid, seastar::compat::source_location l) {
+future<gossiper::endpoint_permit> gossiper::lock_endpoint(inet_address ep, permit_id pid, abortable abortable, seastar::compat::source_location l) {
```

Comment:
I don't think this fix is enough. Even though you add the local state, gossiping will not happen because is_enabled() will return false after aborted. 

I think the proper fix is to abort the gossiper later in the graceful shutdown.

Reply:
> I think the proper fix is to abort the gossiper later in the graceful shutdown.

But still the storage service handlers will fail earlier, and gossiper would abort. Maybe we should allow gossiping and modifying states, just stop firing handlers. Gossip state changes can be merged anyway.

Reply:
> I think the proper fix is to abort the gossiper later in the graceful shutdown.

Makes sense.  Can you please submit an alternative PR with that?
Then we can review it in detail.